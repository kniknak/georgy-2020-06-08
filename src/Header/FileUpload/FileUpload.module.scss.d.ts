declare const styles: {
    readonly input: string;
    readonly button: string;
};
export = styles;