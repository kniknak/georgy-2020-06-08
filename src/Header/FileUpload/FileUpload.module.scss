.input {
    opacity: 0;
    position: absolute;
    z-index: -1;
}

.button {
    cursor: pointer;
    display: flex;
    height: 48px;
    width: 100%;
    justify-content: center;
    align-items: center;
    border: 1px solid gray;
    border-radius: 4px;
    box-sizing: border-box;
}