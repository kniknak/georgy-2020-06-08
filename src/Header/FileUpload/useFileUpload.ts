import{ useCallback, useState } from "react";


export const useFileUpload = (addFile: (file: File) => void) => {
    const [file, setFile] = useState<File | null>(null);

    const onChange: React.ChangeEventHandler<HTMLInputElement> = useCallback(event => {
        if (event.target.files && event.target.files.length) {
            const file = event.target.files[0]
            setFile(file);
            addFile(file)
        }
        event.target.value = "";
    }, [setFile, addFile]);

    const onSubmit: React.FormEventHandler<HTMLFormElement> = useCallback(event => {
        event.preventDefault();

        if (!file) {
            return;
        }

        addFile(file);
    }, [file, addFile]);

    return {
        onSubmit,
        onChange,
    };
};