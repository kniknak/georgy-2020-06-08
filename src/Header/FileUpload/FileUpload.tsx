import React from "react";
import { useFileUpload } from "./useFileUpload";
import * as styles from "./FileUpload.module.scss";

export interface FileUploadProps {
    addFile: (file: File) => void
}

export const FileUpload: React.FC<FileUploadProps> = ({ addFile }) => {
    const {
        onSubmit,
        onChange,
    } = useFileUpload(addFile);

    return (
        <form onSubmit={onSubmit}>
            <input
                className={styles.input}
                type="file"
                onChange={onChange}
                accept="image/png, image/jpeg"
                id="file-upload"
            />
            <label
                className={styles.button}
                htmlFor="file-upload"
            >
                Upload
            </label>
        </form>
    );
};
