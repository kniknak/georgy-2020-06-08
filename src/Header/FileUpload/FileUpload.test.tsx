import React from "react";
import { shallow } from "enzyme";
import { FileUpload, FileUploadProps } from "./FileUpload";
import { useFileUpload } from "./useFileUpload";

jest.mock("./useFileUpload");

const mockedUseFileUpload = useFileUpload as jest.Mock<ReturnType<typeof useFileUpload>>;

describe("Header", () => {
    describe("FileUpload", () => {
        it("renders", () => {
            mockedUseFileUpload.mockReturnValue({
                onSubmit: () => undefined,
                onChange: () => undefined,
            });

            const props: FileUploadProps = {
                addFile: () => undefined,
            };

            const subject = shallow(<FileUpload {...props}/>);

            expect(subject).toMatchSnapshot();

            expect(mockedUseFileUpload).toHaveBeenLastCalledWith(props.addFile);
        });
    });
});