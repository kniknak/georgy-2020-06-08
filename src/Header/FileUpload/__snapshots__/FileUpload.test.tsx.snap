// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header FileUpload renders 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    accept="image/png, image/jpeg"
    id="file-upload"
    onChange={[Function]}
    type="file"
  />
  <label
    htmlFor="file-upload"
  >
    Upload
  </label>
</form>
`;
